lib/Sema/CSDiagnostics.cpp

```diff
@@ -4612,8 +4612,9 @@ bool ArgumentMismatchFailure::diagnoseUseOfReferenceEqualityOperator() const {
   if (getAnchor() == rhs && rhsType->is<FunctionType>()) {
     auto &cs = getConstraintSystem();
     if (cs.hasFixFor(cs.getConstraintLocator(
-            binaryOp, {ConstraintLocator::ApplyArgument,
-                       LocatorPathElt::ApplyArgToParam(0, 0)})))
+            binaryOp,
+            {ConstraintLocator::ApplyArgument,
+             LocatorPathElt::ApplyArgToParam(0, 0, ParameterTypeFlags())})))
```

Comment:
Similar comment regarding the empty parameter flags, though it's less likely the user is overloading the `===` operator. At some point we may want to consider adding a method to `FunctionArgApplyInfo` (and/or `ArgumentMismatchFailure`) that allows you to get a locator for a different arg-to-param application, as `FunctionArgApplyInfo` can retrieve the param flags from the function type.

Reply:
I wrote the code to grab the correct flags inline here for now. I'm not sure how I feel about having a method on FunctionArgApplyInfo that constructs a locator, but maybe it's worth it if this contiinues to come up.

Reply:
I'd say that even a single use warrants abstracting it into a method :)

Reply:
Fair enough :) I named it `getParameterFlagsAtIndex` to distinguish it from the more commonly used `getParameterFlags` which uses the index of the current application.

Reply:
Sounds good!